File: app/lib/library/Socket.dart
import 'dart:async';
import 'package:webmis/env.dart';
import 'package:web_socket_channel/io.dart';
// import 'package:web_socket_channel/status.dart' as status;

/* Socket客户端 */
class Socket{

  // 链接
  static var channel;

  /* 初始化 */
  static Future init(String token) async {
    // Socket
    token = '1fBB/6k3i8cV83M+ld2RFtCZSDmYP9vggwyPhOLHvTKmNxsm1Dz6c0jhYDzwGML9nMozHpim8bTbygAc5S93tS5Q82n8QkLfZ8ZeL/wDpeRzLi8w';
    channel = IOWebSocketChannel.connect(config['socketServer']+'?token='+token);
    // 心跳包
    Timer.periodic(Duration(seconds: 10), (t){
      channel.sink.add('{"type":""}');
    });
    // 监听消息
    // channel.stream.listen((message) {
    //   print(convert.jsonDecode(message));
    // });

  }

  /* Socket 发送 */
  static Future send(String msg) async {
    channel.sink.add('{"type":"newMsg"}');
  }

}

File: python/app/library/Socket.py
from app.Env import Env

import os
import json
from websocket import create_connection

# WebSocket
class Socket:

  # 发送
  def send(self,type,msg):
    url = 'ws://'+Env.socket_ip+':'+str(Env.socket_port)+'/?type='+type+'&token='+Env.key
    msg = json.dumps(msg)
    ws = create_connection(url)
    ws.send(msg)
    ws.close()

  # 发送-Cli
  def sendCli(self,type,msg):
    os.system(Env.cli+' socket send '+type+'  \''+json.dumps(msg)+'\' 2>&1')

